source/s3_meta_request.c

```diff
@@ -2280,76 +2293,129 @@ struct aws_future_void *aws_s3_meta_request_write(
         aws_s3_client_schedule_process_work(meta_request->client);
     }
 
-    return write_future;
+    /* Assert only 1 result field is set (byte_processed could be 0 if data.len is zero) */
+    AWS_ASSERT(
+        (result.is_pending && !result.error_code && !result.bytes_processed) ^
```

Comment:
really minor and no need to fix: i know this works, but mixing logical and bitwise operators feels so wrong here

Reply:
tweaked to hopefully be less confusing